src/NSwag.AspNetCore/SwaggerExtensions.cs

```diff
@@ -98,7 +142,15 @@ public static class SwaggerExtensions
         /// </remarks>
         public static IServiceCollection AddSwagger(this IServiceCollection serviceCollection)
         {
-            return serviceCollection.AddSingleton<IConfigureOptions<MvcOptions>, NSwagConfigureMvcOptions>();
+            serviceCollection.AddSingleton<IConfigureOptions<MvcOptions>, NSwagConfigureMvcOptions>();

```

Comment:
Do you know why this is needed? Doesn't this overwrite/hide client settings?

Reply:
`NSwagConfigureMvcOptions` and the `NSwagMvcApplicationModelConvention` it adds existed before this PR. The convention just changes `ApplicationModel.ApiExplorer.IsVisible` to `true`.

@rynowak in what scenarios would that property change break something?

Reply:
I am a little confused maybe? This was here before.

My understanding of what this does aligns with Dougs. 

That said, with 2.1+ we now enable ApiExplorer visibility by default for `[ApiController]` controller types. https://github.com/aspnet/Mvc/blob/release/2.1/src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs#L64

Reply:
I saw that this was already there but wondered if this construct is a problem somehow?

Reply:
Oh I see. I would suggest updating your convention to look a little bit more like ours https://github.com/aspnet/Mvc/blob/release/2.1/src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs#L61

In particular if the visiblity is null, then you can set it to true. This allows users to opt controllers out of ApiExplorer explicitly.

Reply:
Created new issue: https://github.com/RSuter/NSwag/issues/1671 for later